libraries/gcp_backend.rb

```diff
@@ -223,7 +223,11 @@ def fetch(base_url, template, var_data, request_type = 'Get', body = nil)
       request_type,
       body,
     )
-    return_if_object get_request.send
+    validate_response get_request.send
+  end
+
+  def validate_response(response)
+    response.is_a?(Net::HTTPNotFound) || response.is_a?(Net::HTTPNoContent) ? nil : return_if_object(response)
```

Comment:
do you still need this?

Reply:
yes, for singular resource non-existence usecase we need to return nil to support
 {should_not }  scenarios. Calling return_if_object directly is raising exceptions

Reply:
this is not following proper SRP pattern and we need to handle single & plural connections separately, rather here we are handling them all in one big file which is cumbersome.

Reply:
@balasubramanian-s return_if_object fails if response type is Array.